src/olympia/addons/tasks.py

```diff
@@ -750,6 +751,66 @@ def migrate_legacy_dictionary_to_webextension(addon):
 
 
 @task
+def migrate_webextensions_to_git_storage(ids, **kw):
+    log.info(
+        'Migrating add-ons to git storage %d-%d [%d].',
+        ids[0], ids[-1], len(ids))
+
+    addons = Addon.unfiltered.filter(id__in=ids)
+
+    for addon in addons:
+        # Filter out versions that are already present in the git
+        # storage.
+        versions = addon.versions.filter(git_hash='').order_by('id')
+
+        for version in versions:
+            # Back in the days an add-on was able to have multiple files
+            # per version. That changed, we are very naive here and extracting
+            # simply the first file in the list. For WebExtensions there is
+            # only a very very small number that have different files for
+            # a single version.
+            unique_file_hashes = set([
+                x.original_hash for x in version.all_files
+            ])
+
+            if len(unique_file_hashes) > 1:
+                # Log actually different hashes so that we can clean them
+                # up manually and work together with developers later.
+                log.info(
+                    'Version {version} of {addon} has more than one uploaded '
+                    'file'.format(version=repr(version), addon=repr(addon)))
+
+            if not unique_file_hashes:
+                log.info('No files found for {version} from {addon}'.format(
+                    version=repr(version), addon=repr(addon)))
+                continue
+
+            # Don't call the task as a task but do the extraction in process
+            # this makes sure we don't overwhelm the storage and also makes
+            # sure we don't end up with tasks comitting at random times but
+            # correctly in-order instead.
```

Comment:
Typo `comitting` with one `m`.

You're still going to get a lot of instances of the parent task fired in parallel, right ? Wouldn't it be better to limit concurrency explicitly if you're worried about that ?

Reply:
I'm only concerned about parallel extraction happening for the same add-on, so extraction for various add-ons happening in parallel isn't something I'm too much concerned about. But, looking at this again, I added some rate limiting in c6a9fa67a1 to ensure this is going to be slow but predictable since it doesn't have to be fast anyway.